Display item profit in GE plugin

JoelNarr opened the pull request:
The Grand Exchange plugin already shows the price and the high alch of items in the search field. This change expands that by calculating the profit that would be obtained if the item and the nature rune were bought from the GE for the purpose of alching them. Images below show what the updates look like before and after the change.

![image](https://user-images.githubusercontent.com/51276407/59968756-8e38b680-9504-11e9-9b31-dd1ac9a3a72e.png)
![image](https://user-images.githubusercontent.com/51276407/59968757-9264d400-9504-11e9-97c1-9f88fa1d698a.png)

Adam- commented:
@psikoi can you review this?

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java:
```diff
@@ -225,7 +226,8 @@ private void processResult(List<ItemPrice> result, String lookup, boolean exactM
 				GrandExchangeItemPanel panel = new GrandExchangeItemPanel(item.getIcon(), item.getName(),
-					item.getItemId(), item.getGePrice(), item.getHaPrice(), item.getGeItemLimit());
+					item.getItemId(), item.getGePrice(), item.getHaPrice(), item.getGeItemLimit(),
+					itemManager.getItemPrice(ItemID.NATURE_RUNE));
```
Just look up this price once and store it

psikoi commented:
I really don't like how that interface is getting filled up like that.

I asked about it on discord and abex had a good idea about it.

![image](https://user-images.githubusercontent.com/3278148/60626394-88838080-9de2-11e9-876f-68e575a23c3f.png)

Adam- commented:
I like that, too.

JoelNarr commented:
I reverted changes and added the parenthesis you guys suggested. Below is an image of the new implementation. I saw Adam suggest that this could be a config. I couldn't see an easy way to extend the config manager to where I need to reference it. I'll keep at it if you guys think it's important.
![image](https://user-images.githubusercontent.com/51276407/60739240-d1dce880-9f26-11e9-9b6a-3713b6052081.png)

psikoi commented:
It could be good if the profit/loss was correctly colored (green/red)

JoelNarr commented:
Are you thinking of just the profit/loss portion being colored or the whole string, alch amount and all?

psikoi commented:
Just the difference

JoelNarr commented:
I added a new label so I could color code just the profit margin. Here is the result:
![image](https://user-images.githubusercontent.com/51276407/60937214-2d93d280-a295-11e9-8658-0f6e0f51108f.png)

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java:
```diff
@@ -222,10 +223,12 @@ private void processResult(List<ItemPrice> result, String lookup, boolean exactM
 		SwingUtilities.invokeLater(() ->
 		{
 			int index = 0;
+			int natRune = itemManager.getItemPrice(ItemID.NATURE_RUNE);
```
Small nitpick, but this can be added directly to L231 without assignment to a variable since its value won't be reused.

JoelNarr replied:
conflicts with @Adam review 15 days ago around looking it up once

Nightfirecat replied:
Good call, didn't notice the loop where GrandExchangeItemPanels are initialized. Adam's comment is definitely right to be followed here. 👍

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java:
```diff
@@ -222,10 +223,12 @@ private void processResult(List<ItemPrice> result, String lookup, boolean exactM
 				GrandExchangeItemPanel panel = new GrandExchangeItemPanel(item.getIcon(), item.getName(),
-					item.getItemId(), item.getGePrice(), item.getHaPrice(), item.getGeItemLimit());
+					item.getItemId(), item.getGePrice(), item.getHaPrice(), item.getGeItemLimit(),
+						natRune);
```
This should maintain the indent level of L230.
```suggestion
					natRune);
```

JoelNarr replied:
Looking at it again, there really isn't a reason to not have it on L230, so I moved it up there.

psikoi commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeItemPanel.java:
```diff
@@ -53,7 +53,7 @@
 	private static final Dimension ICON_SIZE = new Dimension(32, 32);
 
 	GrandExchangeItemPanel(AsyncBufferedImage icon, String name, int itemID, int gePrice, Double
```
Why are you not using the primitive "double"?

JoelNarr replied:
I'm not sure why it was originally coded that way, but my guess is to take advantage of the built in intValue() function, which is called in line 156, and I reused in line 164. Going to the implementation, I see that the function just typecasts the variable as int, but I think Double.intValue() is a cleaner look than int(double)

psikoi commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeItemPanel.java:
```diff
@@ -131,27 +134,41 @@ public void mouseReleased(MouseEvent e)
+		alchAndProfitPanel.add(haPriceLabel, BorderLayout.WEST);
+
+		// Profit Margin
+		JLabel profitLabel = new JLabel();
```
I'd personally set all the variables before getting into the UI part, like:

int profitMargin...
Color color...
String string...

JLabel profitLabel...
profitLabel.setText
profitLabel.setForeground

alchAndProfitPanel.add(..)
rightPanel.add(..)

Just so it's clearer, should make it more readable, but that's just my style preference.

JoelNarr replied:
That makes sense to me. Also, having changed it, it is much easier to follow what's happening.

psikoi commented:
The code looks fine, apart from what I commented. I haven't tested yet.

I'm also not very happy with the amount of text that component is displaying. But unless someone has a solution, I guess it makes sense to have the profit margin.
